bucache: save and restore xattrs, ACLs (#1668)
hpc/charliecloud: doc/ch-image.rst

    :code:`ch-image` instances as you want against the same storage directory,
    which risks corruption but may be OK for some workloads.

  :code:`--no-xattrs`

>     Disable xattrs and ALCs from being saved or restored by the build cache.

Reviewer: ```suggestion
    Prevent xattrs and ACLs from being saved or restored by the build cache.
```
?

>     Prevent xattrs and ACLs from being saved or restored by the build cache.

Reviewer: ```suggestion
    Ignore xattrs and ACLs.
```

The build cache uses a relatively new Git under the hood; see the installation
instructions for version requirements. Charliecloud implements workarounds for
Git’s various storage limitations, so things like file metadata, access control
lists, and Git repositories within the image should work. **Important
exception**: No files named :code:`.git*` or other Git metadata are permitted in

> the image’s root directory.

Reviewer: xattrs are addressed in the next paragraph, so maybe revert this?

> `Extended attributes <https://man7.org/linux/man-pages/man7/xattr.7.html>`_

Reviewer: Might be good to mention xattrs for searching purposes, e.g.:

```suggestion
`Extended attributes <https://man7.org/linux/man-pages/man7/xattr.7.html>`_ (xattrs)
```
